CMakeLists.txt

```diff
@@ -1112,7 +1113,7 @@ endif()
 add_executable(       zephyr_prebuilt misc/empty_file.c)
 target_link_libraries(zephyr_prebuilt ${TOPT} ${PROJECT_BINARY_DIR}/linker.cmd ${PRIV_STACK_LIB} ${zephyr_lnk})
 set_property(TARGET   zephyr_prebuilt PROPERTY LINK_DEPENDS ${PROJECT_BINARY_DIR}/linker.cmd)
-add_dependencies(     zephyr_prebuilt ${ALIGN_SIZING_DEP} ${PRIV_STACK_DEP} linker_script offsets)
+add_dependencies(     zephyr_prebuilt ${ALIGN_SIZING_DEP} ${PRIV_STACK_DEP} ${APP_SMEM_DEP} linker_script offsets)
```

Comment:
It is the linker script's that depend on APP_SMEM_DEP, not the elf files.

I believe that the way the dependencies are expressed here might sometimes cause the linker script to be generated before APP_SMEM_DEP has been generated.